jetty-server/src/main/java/org/eclipse/jetty/server/AbstractConnector.java

```diff
@@ -298,6 +302,42 @@ public int getAcceptors()
         return _acceptors.length;
     }
 
+    /**
+     * @return Returns the optional local authority override
+     */
+    @ManagedAttribute("local authority")
+    public HostPort getLocalAuthority()
+    {
+        return _localAuthority;
+    }
```

Comment:
This method should always return a local authority, either one set explicitly or one that is determined dynamically in `doStart` (or `open`).
Then the layers about can just ask for the authority and not have to check if null and then look at the address etc.

Reply:
Also make this a method on `NetworkConnector`

Reply:
This is not a NetworkConnector specific concept.
We have UnixDomainConnector and LocalConnector that can benefit from setLocalAuthority() and are not network connectors.

Reply:
We cannot "always return a local authority".
I tried it.
Going this route breaks expectations in ForwardRequestCustomizer and DetectorConnection and ProxyEndPoint behaviors, so I reverted it.